dataland-frontend/src/components/resources/companiesOnlySearch/CompaniesOnlySearchBar.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <span class="p-fluid">
+    <span class="p-input-icon-left p-input-icon-right">
+      <i class="pi pi-search d-framework-searchbar-input-icon" aria-hidden="true" style="z-index: 20; color: #958d7c" />
+      <AutoComplete
+        inputId="company_search_bar_standard"
+        ref="autocomplete"
+        v-model="searchBarInput"
+        :suggestions="autocompleteArray"
+        :minLength="3"
+        optionLabel="companyInformation.companyName"
+        :autoOptionFocus="false"
+        placeholder="Search company by name or PermID"
+        inputClass="h-3rem d-framework-searchbar-input"
+        panelClass="d-framework-searchbar-panel"
+        style="z-index: 10"
+        @complete="searchCompanyName"
+        @item-select="pushToChooseFrameworkForDataUploadPageForItem"
+      >
+        <template #option="slotProps">
+          <i class="pi pi-search pl-3 pr-3" aria-hidden="true" />
+          <SearchResultHighlighter
+            :text="slotProps.option.companyInformation.companyName"
+            :searchString="latestValidSearchString"
+          />
+        </template>
+      </AutoComplete>
+    </span>
+  </span>
+</template>
+
+<script lang="ts">
+import AutoComplete from "primevue/autocomplete";
+import { StoredCompany } from "@clients/backend";
+import SearchResultHighlighter from "@/components/resources/frameworkDataSearch/SearchResultHighlighter.vue";
+import { defineComponent, inject, ref } from "vue";
+import Keycloak from "keycloak-js";
+import { useRoute } from "vue-router";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import { ApiClientProvider } from "@/services/ApiClients";
+
+export default defineComponent({
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+      autocomplete: ref(),
+    };
+  },
+  name: "ComapniesOnlySearchBar",
+  components: { AutoComplete, SearchResultHighlighter },
+
+  mounted() {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+    this.autocomplete.$refs.focusInput.focus();
+  },
+
+  data: function () {
+    return {
+      searchBarInput: "",
+      latestValidSearchString: "",
+      autocompleteArray: [] as Array<object>,
+      route: useRoute(),
+    };
+  },
+
+  watch: {
+    searchBarInput(newValue: string) {
+      this.saveCurrentSearchStringIfValid(newValue);
+    },
+  },
+
+  methods: {
+    /**
+     * The input string is stored in the variable latestValidSearchString if it is a string and not empty
```

Comment:
If you type a search string into the search bar, then the value of the search string is stored in ``searchBarInput``.
The SearchResultHighlighter component on this page always needs a string.

If you choose one of the suggestions in the dropdown of the search bar by clicking on it or pressing enter on it, then the whole company-object is immediately put into the ``searchBarInput`` variable. That means it is not a string anymore in that case. And that's why you can't directly put ``searchBarInput`` as input prop for the SearcHResultHighlighter (this will lead to an error since it will try execute string-methods on an object).

So to ensure that the SearchResultsHighlighter always receives a string, the current search string (the value of ``searchBarInput``) should only be stored to ``latestValidSearchString``, if it has the type string.
Then we can put ``latestValidSearcHString`` as input for the SearcHResultHighlighter.

Reply:
Ok Thank you